src/print.rs

```diff
@@ -414,31 +415,36 @@ fn all_modules_uniq(module_list: &BTreeSet<String>) -> Vec<String> {
 fn load_formatter_and_modules<'a>(context: &'a Context) -> (StringFormatter<'a>, BTreeSet<String>) {
     let config = &context.root_config;
 
-    let lformatter = StringFormatter::new(&config.format);
-    let rformatter = StringFormatter::new(&config.right_format);
-    let cformatter = StringFormatter::new(&config.continuation_prompt);
-    if lformatter.is_err() {
-        log::error!("Error parsing `format`")
-    }
-    if rformatter.is_err() {
-        log::error!("Error parsing `right_format`")
-    }
-    if cformatter.is_err() {
-        log::error!("Error parsing `continuation_prompt`")
+    let (formatter, config_param) = match &context.target {
+        Target::Main => (StringFormatter::new(&config.format), "format".to_string()),
+        Target::Right => (
+            StringFormatter::new(&config.right_format),
+            "right_format".to_string(),
+        ),
+        Target::Continuation => (
+            StringFormatter::new(&config.continuation_prompt),
+            "continuation_prompt".to_string(),
+        ),
+        Target::Profile(name) => (
+            match config.profiles.get(name) {
+                Some(format) => StringFormatter::new(format),
+                _ => Err(StringFormatterError::Custom("Invalid profile".to_string())),
+            },
+            format!("profile: {}", &name),
+        ),
+    };
+
+    if formatter.is_err() {
+        log::error!("Error parsing `{}`", config_param);
     }
 
-    match (lformatter, rformatter, cformatter) {
-        (Ok(lf), Ok(rf), Ok(cf)) => {
+    match formatter {
+        Ok(lf) => {
             let mut modules: BTreeSet<String> = BTreeSet::new();
             if context.target != Target::Continuation {
                 modules.extend(lf.get_variables());
-                modules.extend(rf.get_variables());
```

Comment:
I think both `lf` and `rf` were added to `modules` here to ensure that modules listed in `right_format` are automatically excluded from `$all` in `$format`. `context.target == Target::Main` might be enough here, but I'm not sure if this is used elsewhere.

Reply:
Ok, now I get why we rendered all formatters 🥲 will see how I can factor better

Reply:
@davidkna I've updated the code, anything else you had in mind?